### PR TITLE
[puz2xd] support block-style cryptics

### LINKED ISSUE #18: Support for cryptic crossword formats?
I don't know if this software supports cryptic puzzles, but I can't seem to make it read any .puz files for cryptic crosswords. Am I holding it wrong, or is it just not a feature? 

Error I see: 
File "................puz2xd.py", line 45, in gen_xd
IndexError: list index out of range

### PR DESCRIPTION
**Pull request by lpesk**
I tested this with the Dec 2009 puzzle here: https://kegler.gitlab.io/Block_style/

### CODE REVIEW

**Comment by anjakefala** [comment]
++ for linking to a sample cryptic!

File: xdplayer/puz2xd.py [hunk 1]
```diff
@@ -27,14 +27,14 @@ def gen_xd(puzfn, clear=True):
     yield ''
 
     def is_block(x, y):
-         return x < 0 or y < 0 or grid[y][x] in BLOCK
+         return x < 0 or y < 0 or x >= p.width or y >= p.height or grid[y][x] in BLOCK
```
Comment (saulpw) [inline xdplayer/puz2xd.py:30]: Where is `p` coming from here?

Comment (lpesk) [reply xdplayer/puz2xd.py:30]: `p = puz_read(puzfn)` at the beginning of `gen_xd()`, same indent level as `def is_block()`. You want it to be an arg here?

Comment (saulpw) [reply xdplayer/puz2xd.py:30]: My fault, I missed the context and didn't realize it was in outer (not global) scope.

**Pull request merged**